secrethub_wrapper.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+/*
+typedef long long time;
+typedef char* uuid;
+
+struct Secret {
+	uuid SecretID;
+	uuid DirID;
+	uuid RepoID;
+	char* Name;
+	char* BlindName;
+	int VersionCount;
+	int LatestVersion;
+	char* Status;
+	time CreatedAt;
+};
+
+struct SecretVersion {
+	uuid SecretVersionID;
+	struct Secret Secret;
+	int Version;
+	char* Data;
+	time CreatedAt;
+	char* Status;
+};
+#include <stdbool.h>
+*/
+import "C"
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+)
+
+// Client creates and returns a new Go client to be used by xgo clients.
+func Client() (*secrethub.Client, error) {
```

Comment:
This causes us to continuously create a new client. This does not necessarily have to be an issue, but for example in the case of using any of the Cloud Identity Providers, this means that the authentication is triggered everytime `Client()` is called. Especially in the way `Client()` is currently called in `ResolveEnv` this can be problematic.

Is there any way in which we can introduce some state to resolve this? In the Go SDK we tackle this by requiring the user to create an instance of the client with `secrethub.NewClient()`. I can imagine that is a little less trivial using swig.

Reply:
Another advantage of re-using the client: we can add caching to the (Golang) SDK to optimize repeating (similar) operations.

Reply:
This was on the list of future improvements, but it is indeed quite problematic in `ResolveEnv`.

I found a way to allow the user to create a new client object and then call these methods on it (and also clean up the client afterwards). However this will take significant effort to implement (~6h maybe). Do we want this for the alpha release? Or would "fixing" the `ResolveEnv` function (making it use one client) also suffice?

Reply:
I implemented the previously mentioned change. The functions are now methods on the `client` object, the underlying go clients are cached and a go client is only created when in the C# code the client constructor is called.